### PR TITLE
Remove force purge tasks autogeneration

### LINKED ISSUE #674: Yorc generates forcePurge tasks on list deployments API endpoint
# Bug Report

## Description

As per https://github.com/ystia/yorc/blob/77cfc4dce12fa16a1e969131d432dc1449962ed4/rest/deployments.go#L386-L393 Yorc will generate a forcePurge tasks for inconsistent deployments.
Sometime those tasks fail to execute and are generated each time the list deployment endpoint is called. This may generate a lot of tasks.

This should be replaced by a warning log allowing an operator to manually remove those inconsistent deployments.

### PR DESCRIPTION
**Pull request by loicalbertin**
# Pull Request description

### What I did

* Removed autogeneration of a force purge task in list deployments endpoint
* Depreciated force purge task
* updated enum lib to allow adding comments on generated enum types

### How to verify it

1. deploy one or more app 
2. remove the deployment status key in consul `consul kv delete _yorc/deployments/<depID>/status`
3. list deployments with the yorc cli
4. should see a log in yorc and inconsistent deployment should remain

### Description for the changelog

* Yorc generates forcePurge tasks on list deployments API endpoint ([GH-674](https://github.com/ystia/yorc/issues/674))

## Applicable Issues

* Fixes #674 
* Backported to release/4.0 branch in #676

### CODE REVIEW

File: rest/deployments.go [hunk 1]
```diff
@@ -366,35 +364,26 @@ func (s *Server) getDeploymentHandler(w http.ResponseWriter, r *http.Request) {
...
 			if deployments.IsDeploymentNotFoundError(err) {
-				// Deployment is not found : we force deletion and ignore it
-				go func() {
-					log.Debugf("Force purge inconsistent deployment with ID:%q", deploymentID)
-					if _, err := s.tasksCollector.RegisterTask(deploymentID, tasks.TaskTypeForcePurge); err != nil {
-						log.Printf("Failed to force purge deployment with ID:%q due to error:%+v", deploymentID, err)
-					}
-				}()
+				// Deployment is not found : we force rapport error and ignore it
+				log.Printf("[WARNING] deployment %q is inconsistent, igoring it from deployments list. You please investigate and retport this issue.")
```
Comment (laurentganne) [inline rest/deployments.go:383]: Missing deployement ID in argument of Printf

Comment (loicalbertin) [reply rest/deployments.go:383]: good catch

File: tasks/structs.go [hunk 2]
```diff
@@ -16,29 +16,35 @@ package tasks
...
+Deploy
+UnDeploy
+ScaleOut
+ScaleIn
+Purge
+CustomCommand
+CustomWorkflow
+Query
+Action
+ForcePurge // ForcePurge is deprecated and should not be used anymore this stay here to prevent task renumbering colision
```
Comment (laurentganne) [inline tasks/structs.go:31]: ForcePurge is still used in rest/deployments.go line 310 in deleteDeploymentHandler().
Should it be replaced by a warning there as well ?

Comment (loicalbertin) [reply tasks/structs.go:31]: Well, you are right I did not see this.
But it seems to be a valid use case here for force purge... WDYT?
This should be refactored later as we discussed IMHO.

Comment (laurentganne) [reply tasks/structs.go:31]: Yes, ok !

**Review by laurentganne — APPROVED** [review]
Looks good!

**Pull request merged**